api/src/devcomp/domain/models/passage-events/PassageElementEvent.js

```diff
@@ -0,0 +1,27 @@
+import { assertHasUuidLength, assertNotNullOrUndefined } from '../../../../shared/domain/models/asserts.js';
+import { PassageElementEventInstantiationError } from '../../errors.js';
+import { PassageEvent } from './PassageEvent.js';
+
+/**
+ * @abstract PassageElementEvent
+ * After the write operation is successful, the system generates events that describe what changed (like "Passage started" or "Video started").
+ * These events serve as notifications about the updates.
+ * See Event sourcing pattern for more information.
+ * https://martinfowler.com/eaaDev/EventSourcing.html
+ *
+ * This is the base class for all PassageElementEvent. Subclasses should be named in past tense.
```

Comment:
```suggestion
 * This is the base class for all PassageElementEvent. It's syntaxic sugar for a `PassageEvent` that contains an element, it exists because lots of events happen on a element that we store in the `data` key.. Subclasses should be named in past tense.
```